\section*{Conclusions}
In this work, we proposed a framework for LV phenotyping based on unsupervised geometric learning techniques on image-derived 3D meshes, for the purpose of discovering genetic variations that affect LV shape through GWAS. The proposed methodology is based on finding a latent low-dimensional representation of the CMR-derived LV 3D meshes using convolutional mesh-autoencoders and then performing GWAS on the learned latent features. In contrast with previous efforts carried out to obtain attributes from meshes to perform genetic discoveries, we did not look to explictly enforce an association between the latent representation and the genotype, but instead sought to enforce the learning of uncorrelated latent variables, representative of shape characteristics. % This allowed us to assume a uniform distribution of GWAS $p$-values under the null hypothesis of no genetic effect.
As was hypothesised, this dimensionality reduction method, using Kullback-Leibler regularisation, yielded phenotypes with significantly associated SNPs. 
Two phenotypes were discovered that showed genome-wide significant associations. One of them linked to LV sphericity, and the associated genetic locus was mapped to gene PLN with high confidence. Mutations in this gene have been reported to be a cause for dilated cardiomyopathy. Also, it has been recently reported for the first time to be associated with LV cavity volume. 
These results validate our methodology as a way to extract knowledge about the genetics driving the morphology of organs, leveraging databases that provide linked genetic and imaging data, such as the UKB.